Week1/Section3/Exercise7/Source7.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Exercise 7 - Lottery
+ * Write a program that will simulate the Lottery. Create two arrays of 5 integers.
+ * The first array should be populated with 5 random numbers from 1-20. For the second array, ask the user to input 5 numbers from 1-20.
+ * Make sure to handle invalid input and make sure numbers are not repeated.
+ * For each number that is matched in the same index, give the user $10.
+ */
+
+#include <iostream>
+
+int main()
+{
+	int Lottery[5];
+	int UserGuess[5];
+
+	std::cout << "-----> Welcome to the Teravision Lottery <-----\n"
+			  << "   	may the luck be with you!" << std::endl;
+
+	/* initialize random seed: */
+	srand(time(NULL));
+
+	for (int i = 0; i < sizeof(Lottery) / sizeof(int); i++)
+	{
+		// populate the Lottery array
+		int RandomNumber = 1 + rand() % 20;
+		bool isRandomRepeated = false;
+
+		for (int x : Lottery)
+		{
+			if (x == RandomNumber)
+			{
+				isRandomRepeated = true;
+				i--;
+			}
+		}
+
+		if (!isRandomRepeated)
+		{
+			Lottery[i] = RandomNumber;
+		}
+	}
+
+	int index = 0;
+	while (index < (sizeof(Lottery) / sizeof(int)))
+	{
+		int UserInput;
+		bool isRepeated = false;
+		std::cout << "Enter your guess between [1, 20]: ";
+		std::cin >> UserInput;
+
+		// checks if the number is repeated
+		for (int x : UserGuess)
```

Comment:
Stuff like this is a good place to start implementing functions :) What about a `bool IsNumberRepeated()` function? That takes in an array, its size, and a number to check?

Reply:
Nice! fixed